### ISSUE TITLE
Pause button

### ISSUE THREAD

**Issue by BeckeBauer** [kevinpapst/kimai2#963]
**Is your feature request related to a problem? Please describe.**
Very often, I need to pause a time-record (incoming calls etc). Then, I hit the play button in order to hit the stop button. 
![grafik](https://user-images.githubusercontent.com/51003649/61226022-00767280-a722-11e9-893c-cd2a6efd78e6.png)
After the break, I go to the drop-down menu "Actions" and choose "start again".

**Describe the solution you'd like**
To have a second button "pause" 
![grafik](https://user-images.githubusercontent.com/51003649/61225550-159ed180-a721-11e9-8b4f-9887fca7fe63.png)
next to the "stop button" would do the trick. If one hits this button, it stops the recording (like hitting the stop button but the entry is still be shown if one clicks the play button). If one hits the pause button again, the time recording starts over (similar to the "start again" action).

**Comment by rahulkrishnans** [comment]
We have completely migrated to kimai time tracker and using the API, we grab the duration of tasks for performance evaluation. In database, kimai builds unique ids for each record, so we are referring this id to link other data in other server. The problem we faced is that when a user stops the task/activity to take interval and restarts the same task, the ids are changing and as a result we cannot track the whole time spent for that particular task/activity by that user in that particular day.

Would it be possible to add an extra column in database named "Task id" (or any name which you think suitable) which will populate a unique id just like the record id? When a button like "Pause" is pressed which is placed inside "Actions" button, the time will be stopped. When we press "Start again", kimai checks whether the previous task is paused or not and if it is paused, timer should start with the recent task id.

Please check duration of **Product A** in the below data. 

For example: 

**Current data**

_Kimai server_             
```
Record_id  Date        Duration  Customer    Project  Activity   Tags        User
1234       2019-12-13  1200      C1          P1       A1         Product A   User A
1235       2019-12-13  1400      C1          P1       A1         Product A   User A
1236       2019-12-13  1300      C1          P1       A1         Product B   User B
1237       2019-12-14   800      C1          P2       A1         Product C   User A
1238       2019-12-14  1400      C1          P1       A1         Product A   User A
```

_Performance server (our in-house server)_
```
Product    Date        Output  Record_id  Duration
Product A  2019-12-13   O1     1235   	  1400
Product B  2019-12-13   O2     1236   	  1300
Product C  2019-12-14   O3     1237    	  800
Product A  2019-12-14   1O     1238    	  1400

```

**Required data**

_Kimai server_             
```
Record_id  Date        Task_id  Duration  Customer    Project  Activity   Tags        User     Status 
1234       2019-12-13   5342     1200      C1          P1       A1         Product A  User A   pause
1235       2019-12-13   5342     1400      C1          P1       A1         Product A  User A   incomplete
1236       2019-12-13   5343     1300      C1          P1       A1         Product B  User B   finished
1237       2019-12-14   5344      800      C1          P2       A1         Product C  User C   finished
1238       2019-12-14   5345     1400      C1          P1       A1         Product A  User A   finished
```

_Performance server (our in-house server)_
```
Product    Date        Output  Task_id  Duration
Product A  2019-12-13   O1     5342      2600
Product B  2019-12-13   O2     5343      1300
Product C  2019-12-14   O3     5344       800
Product A  2019-12-14   1O     5345      1400
```

**Comment by vic-t** [comment]
Realizing too late there is an open feature request for this, I started a new discussion on the discussion board, see https://github.com/kevinpapst/kimai2/discussions/2365#discussion-3224940

**Comment by sollermun** [comment]
I've already voted for this feature, but want to add my two cents.

In my line of work, I like to keep contemporaneous time records to the extent possible. I also charge in increments of 1/10 per hour and utilize the rounding feature in Kimai. Unfortunately, it is unavoidable that I am interrupted by an important telephone call or there is some other distraction that requires me to stop what I am doing momentarily. 

Ideally, there would be an ability to pause what I am working on (without rounding), start and stop a clock on a different matter, and then resume work where I left off on the original task.

Right now, stopping in the middle of a task results in multiple rounding of time and two or more entries for the same task, which requires further work in editing time entries or invoices.

I think this is an essential feature that would benefit nearly everyone that uses Kimai.

**Comment by vic-t** [comment]
Apart from Kimai, I worked with two other solutions for managing project times: Gleeo and Clockify. Both are commercial and proprietary products but both have free versions for small teams. After having worked with Kimai for a little over a year, I'm currently working with Clockify because it comes with a superior desktop app and still allows me to work with custom fields. I'm hopeful to switch back to Kimai in the future but for now I don't want to miss the convenience of a modern desktop app. 

Gleeo's approach is basically what is described in this feature request, that is, after a task has been started, the user can choose wether he wants to stop the task or just pause it. In case a task is paused, it can be picked up later. Multiple tasks can be created and paused. I think this approach makes sense in situations where it is important to know how much total time was spent on a certain task rather than understanding at what specific times it was worked on. 

Clockify has a different but also valid approach that may be easier to implement in Kimai. There is no pause button, a running task can only be stopped. However, a stopped task can be resumed in that it is used as a template for a new task. So, resuming a stopped task will create a separate entry but as long as the task name and description are the same, reporting will merge the two separate time entries and create one. Rounding only happens to the merged entry. 

To be clear, in order to resume a task, it is not mandatory to select the task to be resumed - just creating a new task, on the same day, with the same title and description will have the same effect. 

BTW, you may say that typing the title and description over again is tedious. I agree. The value add Clockify provides here is by letting me type just part of the title and it will suggest to fill the rest automatically. If I'm fine with the suggestion, I can just click enter and the task will be resumed. If I overwrite it with something else, a new one will be created. 

I hope this helps.

**Comment by kevinpapst** [comment]
Yes, in my opinion it is always super helpful to compare the product of a 20+ persons team from a venture capital funded company, with a fucking free and open-source software, built by one person and from his own money. ❤️  it!

**Comment by vic-t** [comment]
@kevinpapst Why the childish and hateful response towards a paying customer who took his time to provide input for you? 

Gleeo used to be a one-person shop, maybe still is. I used it before switching to Kimai because it lacked the possibility to mark exported entries as such. But it does everything else well. 

Since we moved to a different ERP in the meantime, I had to re-evaluate our use case and came to the conclusion that we really need a desktop app. Unfortunately, there is nothing available for Kimai that we liked and, being a 2 person team, we have no resources or finances to develop one of our own. 

So, we found Clockify and it works for us. I'd still prefer to have full control of my data which is why I go back to Kimai regularly to check what's new. 

At any rate, my post wasn't meant to put Kimai in a bad light but just to provide ideas how the pause feature could be done.

**Comment by kevinpapst** [comment]
> Why the childish and hateful response towards a paying customer who took his time to provide input for you?

This is trolling and my last warning.

Providing input is great. Posting that another product is better and that you switched over is not and it does not contribute any value to the discussion. Especially when you compare products with a totally different background. Leave it out of here. 

You can install Kimai like any other WebApp from Google Chrome. 
<img width="545" alt="Bildschirmfoto 2023-05-11 um 16 04 38" src="https://github.com/kimai/kimai/assets/533162/4e2d4a38-b49e-471c-b995-ecc007a978d4">
<img width="1527" alt="Bildschirmfoto 2023-05-11 um 16 04 45" src="https://github.com/kimai/kimai/assets/533162/b276a7f4-0e8b-47df-b810-57c5d7a3cab3">

I don't want any reply. This thread is about a Pause button.

**Comment by vic-t** [comment]
You're "warning" me? Kevin, have you lost it? Put your money where your mouth is and refund my purchases right now. 

We are in software development ourselves, albeit not in the open source space. We always look at what our competitors do better than we do. Even if they have tons of more money than we do and we may not have the same capabilities, we will still strive to make our product better. I did nothing else here I wouldn't have done with my own team, i.e. show how the competition does it as an inspiration. 

Absolutely unnecessary, petty comments on your part because of a bruised ego. Deeply disappointed. 

Unsubscribing from your project and all threads, will not spend another minute on this.

**Comment by jcn50** [comment]
Wow: 4 years later and still no sign of this feature...

**Comment by kevinpapst** [comment]
**Everyone can vote via reactions on the initial idea from @BeckeBauer**

- If you think this feature is vital for your organization, you are free to talk to your boss and organize the funding.
- If you are a developer and think you know how to handle break times properly and in depth, please sent in a Pull request with your code.
- If you think you have vital input how this should work, create a discussion.

---

@jcn50 no drunken posts please!

> There is one simple rule in our "Code of conduct": Don't be an ass!

But I have to thank you anyhow, I needed this to finally lock down this issue to prevent further bullshit or hate comments.
